src/compiler/program.ts

```diff
@@ -4229,6 +4229,15 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
             }
         }
 
+        if (options.isolatedDeclarations) {
+            if (getAllowJSCompilerOption(options)) {
+                createDiagnosticForOptionName(Diagnostics.Option_0_cannot_be_specified_with_option_1, "allowJs", "isolatedDeclarations");
+            }
+            if (!options.declaration) {

```

Comment:
```suggestion
            if (!getEmitDeclarations(options)) {

```
since `declaration` is implied by some other options.